poetry/core/packages/__init__.py

```diff
@@ -63,8 +63,8 @@ def dependency_from_pep_508(
     name = parts[0].strip()
     if len(parts) > 1:
         rest = parts[1]
-        if ";" in rest:
-            name += ";" + rest.split(";", 1)[1]
+        if " ;" in rest:
+            name += " ;" + rest.split(" ;", 1)[1]
```

Comment:
```suggestion
        if ";" in rest:
            name += " ;" + rest.split(";", 1)[1].strip()
```

Reply:
This won't work if the url contains `;`.